src/libtiled/hexagonalrenderer.cpp

```diff
@@ -289,6 +318,9 @@ void HexagonalRenderer::drawTileLayer(QPainter *painter,
 
     CellRenderer renderer(painter, CellRenderer::HexagonalCells);
 
+    const int endX = map()->infinite() ? layer->bounds().width() : layer->width();
```

Comment:
What about `layer->width()`? My point is that `endX` should be the right-most X coordinate (+1 because it's one beyond that), like it is for finite maps. But you just took `bounds().width()`, which does not take into account that `bounds().x()` isn't always 0.